### PR TITLE
Terratest plus 12

### PR DESCRIPTION
**Pull request by kilbergr**
https://www.pivotaltracker.com/story/show/169410322: introduce Terratest into the module. The PR also includes:

Makefile - For running pre-commit and Terratest
golangci-lint - Go linting
dependabot - To keep go modules up to date
README - Updates for how to run tests and new tf version
versions - Specify tf version

The Terratest example lives in examples/simple. Running against our CI AWS account returns

--- PASS: TestTerraformAwsIamEcrCi (18.70s)
PASS
ok  	terraform-aws-iam-ecr-ci/test	18.714s

### CODE REVIEW

File: examples/simple/main.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module "ecr_ci" {
+  source = "../../"
+
+  ecr_repo = "${aws_ecr_repository.main.name}"
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
  ecr_repo = aws_ecr_repository.main.name
```

File: examples/simple/main.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  ci_name  = "${var.ci_name}"
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
  ci_name  = var.ci_name
```

File: examples/simple/main.tf [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+}
+
+resource "aws_ecr_repository" "main" {
+  name = "${var.ecr_repo}"
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
  name = var.ecr_repo
```

File: examples/simple/main.tf (hunk 1 above)
Comment (kilbergr) [reply examples/simple/main.tf]: I resolved this all in 1 pr so I wouldn't have 3 commits idk if that was the "right" way but that's what I did

**Pull request merged**